iceoryx_posh/test/moduletests/test_popo_active_call_set.cpp

```diff
@@ -0,0 +1,940 @@
+
+    for (auto& e : events)
+    {
+        e.triggerStoepsel();
+    }
+
+    // 10 times more callback runs allowed to allow potential overtriggering
+    unblockTriggerCallback(10U * iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET);
+    std::this_thread::sleep_for(std::chrono::milliseconds(CALLBACK_WAIT_IN_MS));
+
+    TIMING_TEST_EXPECT_TRUE(g_triggerCallbackArg[0].m_source == &events[0U]);
+    TIMING_TEST_EXPECT_TRUE(g_triggerCallbackArg[0].m_count == 2U);
+    for (uint64_t i = 1U; i < iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET; ++i)
+    {
+        TIMING_TEST_EXPECT_TRUE(g_triggerCallbackArg[i].m_source == &events[i]);
+        TIMING_TEST_EXPECT_TRUE(g_triggerCallbackArg[i].m_count == 1U);
+    }
+});
+
+TIMING_TEST_F(ActiveCallSet_test, TriggeringAllEventsCallsAllCallbacksOnce, Repeat(5), [&] {
+    std::vector<SimpleEventClass> events(iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET);
+
+    AttachEvent<iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET - 1>::doIt(
+        *m_sut, events, SimpleEvent::StoepselBachelorParty);
+
+    activateTriggerCallbackBlocker();
+    events[0U].triggerStoepsel();
+    std::this_thread::sleep_for(std::chrono::milliseconds(CALLBACK_WAIT_IN_MS));
+
+    for (auto& e : events)
+    {
+        e.triggerStoepsel();
+    }
+
+    // 10 times more callback runs allowed to allow potential overtriggering
+    unblockTriggerCallback(10U * iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET);
+    std::this_thread::sleep_for(std::chrono::milliseconds(CALLBACK_WAIT_IN_MS));
+
+    events[0U].triggerStoepsel();
+    std::this_thread::sleep_for(std::chrono::milliseconds(CALLBACK_WAIT_IN_MS));
+
+    TIMING_TEST_EXPECT_TRUE(g_triggerCallbackArg[0].m_source == &events[0U]);
+    TIMING_TEST_EXPECT_TRUE(g_triggerCallbackArg[0].m_count == 3U);
+    for (uint64_t i = 1U; i < iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET; ++i)
+    {
+        TIMING_TEST_EXPECT_TRUE(g_triggerCallbackArg[i].m_source == &events[i]);
+        TIMING_TEST_EXPECT_TRUE(g_triggerCallbackArg[i].m_count == 1U);
+    }
+});
+//////////////////////////////////
+// END
+//////////////////////////////////
+
+//////////////////////////////////
+// BEGIN concurrent attach / detach
+//////////////////////////////////
+TIMING_TEST_F(ActiveCallSet_test, AttachingWhileCallbackIsRunningWorks, Repeat(5), [&] {
+    std::vector<SimpleEventClass> events(iox::MAX_NUMBER_OF_EVENTS_PER_ACTIVE_CALL_SET);
+
+    m_sut->attachEvent(events[0U], SimpleEvent::StoepselBachelorParty, triggerCallback<0U>);
+
+    g_triggerCallbackRuntimeInMs = 3U * CALLBACK_WAIT_IN_MS / 2U;
+    events[0U].triggerStoepsel();
+    std::this_thread::sleep_for(std::chrono::milliseconds(CALLBACK_WAIT_IN_MS));
+
+    m_sut->attachEvent(events[1U], SimpleEvent::StoepselBachelorParty, triggerCallback<1U>);
+    events[1U].triggerStoepsel();
+    std::this_thread::sleep_for(std::chrono::milliseconds(CALLBACK_WAIT_IN_MS * 2U));
```

Comment:
No since we would then end up in a deadlock! The idea here is somehow to test that here no race condition occurs - it is a very basic test and actually we require a very extensive stress test for this but for now this should sufficn.

Reply:
This could also cause a deadlock - should be addressed in a stress test.